go/engine/pgp_warnings_test.go

```diff
@@ -0,0 +1,519 @@
+		require.NoErrorf(t, clearsignInput.Close(), "finishing clearsign [%s]", e.Name)
+		arg := &PGPVerifyArg{
+			Source:   clearsignSink,
+			SignedBy: signedBy,
+		}
+		eng := NewPGPVerify(alice.tc.G, arg)
+		require.NoErrorf(t, RunEngine2(m, eng), "engine failure [%s]", e.Name)
+		require.Lenf(t, eng.SignatureStatus().Warnings, e.Count, "warnings count [%s]", e.Name)
+
+		// Then process the attached sig
+		attachedInput, _, err := libkb.ArmoredAttachedSign(
+			attachedSink,
+			*bob.key.Entity,
+			nil,
+			cfg,
+		)
+		require.NoErrorf(t, err, "attached sign failure [%s]", e.Name)
+		_, err = attachedInput.Write([]byte(pgpWarningsMsg))
+		require.NoErrorf(t, err, "writing to attached signer [%s]", e.Name)
+		require.NoErrorf(t, attachedInput.Close(), "writing to attached sign [%s]", e.Name)
+		arg = &PGPVerifyArg{
+			Source:   attachedSink,
+			SignedBy: signedBy,
+		}
+		eng = NewPGPVerify(alice.tc.G, arg)
+		require.NoErrorf(t, RunEngine2(m, eng), "engine failure [%s]", e.Name)
+		require.Lenf(t, eng.SignatureStatus().Warnings, e.Count, "warnings count [%s]", e.Name)
+
+		// Detached signatures are probably the easiest
+		require.NoError(t, openpgp.ArmoredDetachSignText(
+			detachedSink,
+			bob.key.Entity,
+			strings.NewReader(pgpWarningsMsg),
+			cfg,
+		), "detached sign failure")
+		arg = &PGPVerifyArg{
+			Source:    strings.NewReader(pgpWarningsMsg),
+			Signature: detachedSink.Bytes(),
+			SignedBy:  signedBy,
+		}
+		eng = NewPGPVerify(alice.tc.G, arg)
+		require.NoErrorf(t, RunEngine2(m, eng), "engine failure [%s]", e.Name)
+		require.Lenf(t, eng.SignatureStatus().Warnings, e.Count, "warnings count [%s]", e.Name)
+
+		return
+	}
+
+	if e.Mode == "decrypt" {
+		// Mostly the same as decrypt, except we're using different code paths
+		// to achieve roughly the same effect.
+
+		var (
+			clearsignSink       = libkb.NewBufferCloser()
+			clearsignOutputSink = libkb.NewBufferCloser()
+			attachedSink        = libkb.NewBufferCloser()
+			attachedOutputSink  = libkb.NewBufferCloser()
+		)
+
+		var signedBy string
+		if e.Known {
+			signedBy = bob.user.Username
+		}
+
+		// Start with the clearsign sig, which technically isn't even something
+		// decryptable.
+		clearsignInput, err := clearsign.Encode(
+			clearsignSink,
+			bob.key.PrivateKey,
+			cfg,
+		)
+		require.NoErrorf(t, err, "clearsign failure [%s]", e.Name)
+		_, err = clearsignInput.Write([]byte(pgpWarningsMsg))
+		require.NoErrorf(t, err, "writing to clearsign [%s]", e.Name)
+		require.NoErrorf(t, clearsignInput.Close(), "finishing clearsign [%s]", e.Name)
+		arg := &PGPDecryptArg{
+			Sink:         clearsignOutputSink,
+			Source:       clearsignSink,
+			AssertSigned: true,
+			SignedBy:     signedBy,
+		}
+		eng := NewPGPDecrypt(alice.tc.G, arg)
+		require.NoErrorf(t, RunEngine2(m, eng), "engine failure [%s]", e.Name)
+
+		// TODO: Y2K-1334 Fix this test
+		//require.Lenf(t, eng.SignatureStatus().Warnings, e.Count, "warnings count [%s]", e.Name)
```

Comment:
```suggestion
		require.Lenf(t, eng.SignatureStatus().Warnings, e.Count, "warnings count [%s]", e.Name)
```
we arrive at correct warning list here, just the output text has extra newline.